alf/examples/lm_conf.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A simple language modeling example using Transformer.
+
+You can use the following command to run it:
+
+.. code-block:: bash
+
+    python -m alf.bin.train --conf lm_conf.py --root_dir ~/tmp/lm/exp0
+
+Note: You need to first install torchtext using ``pip install torchtext``
+
+"""
+from functools import partial
+import math
+import torch
+import torch.nn as nn
+from typing import Callable
+
+import alf
+from alf import layers, networks
+from alf.algorithms.config import TrainerConfig
+from alf.algorithms.algorithm import Algorithm, LossInfo
+from alf.utils import common
+
+
+@alf.configurable
+def create_model(ntokens,
+                 embedding_dim=200,
+                 memory_size=128,
+                 num_layers=4,
+                 num_heads=4):
+    embedding_layer = torch.nn.Embedding(ntokens, embedding_dim)
+    embedding_layer.weight.data.uniform_(-0.1, 0.1)
+    return networks.Sequential(
+        embedding_layer,
+        layers.Reshape(1, -1),
+        networks.TransformerNetwork(
+            input_tensor_spec=alf.TensorSpec((1, embedding_dim)),
+            num_prememory_layers=0,
+            num_attention_heads=num_heads,
+            d_ff=4 * embedding_dim,
+            core_size=1,
+            use_core_embedding=False,
+            memory_size=memory_size,
+            num_memory_layers=num_layers,
+            centralized_memory=False),
+        layers.FC(embedding_dim, ntokens, kernel_init_gain=0.),
+        input_tensor_spec=alf.TensorSpec((), dtype=torch.int64),
+    )
+
+
+@alf.configurable
+class LMAlgorithm(Algorithm):
```

Comment:
Should put this class into an algorithm file instead?

Reply:
Since this is just an example, I don't feel it is worth to put into algorithms.